lib/experimental/modules/class-gutenberg-modules.php

```diff
@@ -90,27 +116,25 @@ public static function print_import_map() {
 	 * Prints all the enqueued modules using <script type="module">.
 	 */
 	public static function print_enqueued_modules() {
-		foreach ( self::$enqueued as $module_identifier ) {
-			if ( isset( self::$registered[ $module_identifier ] ) ) {
-				$module = self::$registered[ $module_identifier ];
-				wp_print_script_tag(
-					array(
-						'type' => 'module',
-						'src'  => $module['src'] . self::get_version_query_string( $module['version'] ),
-						'id'   => $module_identifier,
-					)
-				);
-			}
+		foreach ( self::get_enqueued() as $module_identifier => $module ) {
+			wp_print_script_tag(
+				array(
+					'type' => 'module',
+					'src'  => $module['src'] . self::get_version_query_string( $module['version'] ),
+					'id'   => $module_identifier,
+				)
+			);
 		}
 	}
 
 	/**
-	 * Prints the link tag with rel="modulepreload" for all the static
-	 * dependencies of the enqueued modules.
+	 * Prints the the static dependencies of the enqueued modules using <link rel="modulepreload">.
 	 */
-	public static function print_module_preloads() {
-		foreach ( self::get_dependencies( self::$enqueued, array( 'static' ) ) as $dependency_identifier => $module ) {
-				echo '<link rel="modulepreload" href="' . $module['src'] . self::get_version_query_string( $module['version'] ) . '" id="' . $dependency_identifier . '">';
+	public static function print_preloaded_modules() {
+		foreach ( self::get_dependencies( array_keys( self::get_enqueued() ), array( 'static' ) ) as $module_identifier => $module ) {
+			if ( true !== $module['enqueued'] ) {
+				echo '<link rel="modulepreload" href="' . $module['src'] . self::get_version_query_string( $module['version'] ) . '" id="' . $module_identifier . '">';
```

Comment:
while I see that this was already here before your change, it would be good to escape these values to avoid HTML syntax issues. can be done with `esc_attr()` or with the Tag Processor.

```php
echo sprintf(
	'<link rel="modulepreload" href="%s" id="%s">',
	esc_attr( $module['src'] . self::get_version_query_string( $module['version'] ) ),
	esc_attr( $module_identifier )
);
```

Reply:
Totally forgot about these. Thanks Dennis! Done in https://github.com/WordPress/gutenberg/pull/57231/commits/9aeb4997a97b38dbfbc72d7df43f51a46ec0f369.